alpha/apps/kaltura/lib/live/kLiveStreamEventConsumer.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+
+class kLiveStreamEventConsumer implements kObjectChangedEventConsumer
+{
+    const LIVE_STREAM_VOD_THUMBNAIL_TAG = 'live_entry_vod';
+    /**
+     * @inheritDoc
+     */
+    public function objectChanged(BaseObject $object, array $modifiedColumns)
+    {
+        if ($object instanceof entry)
+        {
+            $this->handleEntryChanged($object, $modifiedColumns);
+        }
+
+        if ($object instanceof asset)
+        {
+            $this->handleAssetChanged($object);
+        }
+
+        return true;
+    }
+
+    protected function handleAssetChanged (asset $object)
+    {
+        if (!($object instanceof thumbAsset))
+        {
+            KalturaLog::info('The object being handled is not a ThumbAsset. Skipping');
+            return true;
+        }
+        /* @var $object thumbAsset */
+        $entry = entryPeer::retrieveByPK($object->getEntryId());
+        if (!$entry)
+        {
+            KalturaLog::info("Thumb asset entry ID {$object->getEntryId()} could not be retrieved. Skipping");
+            return true;
+        }
+
+        if ($entry instanceof LiveEntry)
+        {
+            $recordedEntry = entryPeer::retrieveByPK($entry->getRecordedEntryId());
+            if (!$recordedEntry)
+            {
+                KalturaLog::info("Recorded entry ID for live entry {$entry->getId()} could not be retrieved. Skipping.");
+                return true;
+            }
+
+            $object->copyToEntry($recordedEntry->getId());
+        }
+
+        if($entry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE){
+             $liveThumbAsset = $object->copyToEntry($entry->getRootEntryId());
+             $liveThumbAsset->setTags(self::LIVE_STREAM_VOD_THUMBNAIL_TAG);
+             $liveThumbAsset->save();
+        }
+
+        return true;
+    }
+
+    protected function handleEntryChanged(entry $object, array $modifiedColumns)
+    {
+        if (!($object instanceof LiveEntry) || !($object->getRecordStatus()))
+        {
+            KalturaLog::info("Entry {$object->getId()} is either not a live entry, or does not have recording enabled. Skipping.");
+            return true;
+        }
+
+        if(isset($modifiedColumns[kObjectChangedEvent::CUSTOM_DATA_OLD_VALUES]['']['recorded_entry_id'])) // array_key_exists
```

Comment:
a. we can. The question then becomes "why doesn't the LiveEntry class itself use this constant?" https://github.com/kaltura/server/blob/Propus-16.18.0/alpha/lib/model/LiveEntry.php#L256
b. because the recorded_entry_id is not a column. It's part of the custom_data on the entry object.

Reply:
@MosheMaorKaltura  changed to us constant. Should I fix the hard-coded string in LiveEntry as part of this PR as well?